src/Microsoft.DotNet.CodeFormatting/FormattingEngineImplementation.cs

```diff
@@ -196,7 +205,43 @@ public ImmutableArray<DiagnosticDescriptor> AllSupportedDiagnostics
             var watch = new Stopwatch();
             watch.Start();
 
-            var diagnostics = await GetDiagnostics(project, cancellationToken).ConfigureAwait(false);
+            var workspace = project.Solution.Workspace;
+
+            await FormatProjectWithSyntaxAnalyzersAsync(workspace, project.Id, cancellationToken);
+            await FormatProjectWithLocalAnalyzersAsync(workspace, project.Id, cancellationToken);
+            await FormatProjectWithGlobalAnalyzersAsync(workspace, project.Id, cancellationToken);
+
+            watch.Stop();
+            FormatLogger.WriteLine("Total time for formatting {0} - {1}", project.Name, watch.Elapsed);
+        }
+
+        private async Task FormatProjectWithSyntaxAnalyzersAsync(Workspace workspace, ProjectId projectId, CancellationToken cancellationToken)
+        {
+            var analyzers = _analyzers.Where(a => a.SupportedDiagnostics.All(d => d.CustomTags.Contains(RuleType.Syntactic)));
```

Comment:
Yup, this is what I was worried about. This means that unless an analyzer explicitly declares its scope (`RuleType`), CodeFormatter won't run it at all. We had originally hoped to be able to pass arbitrary analyzer DLLs on the codeformatter command line so end users could say "please format my project with these style analyzers", but with this design, only analyzers that know enough to set these particular custom tags can run.

One option is to follow this up by implementing command line options like
```
--analyzer-assemblies foo.dll,bar.dll --rules SA1001,SA1002
```
or
```
--analyzer-assemblies foo.dll,bar.dll --all-rules
```
... which would allow the user to say "I know what I'm doing, just run these".